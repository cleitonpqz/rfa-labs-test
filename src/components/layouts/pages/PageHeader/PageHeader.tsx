import CompanyIdentifier from "components/library/CompanyIdentifier/CompanyIdentifier";
import ProfileCard from "components/library/ProfileCard/ProfileCard";
import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  height: 112px;
  background: #ffffff;

  border: 0.5px solid #ebebf9;
  box-sizing: border-box;

  box-shadow: 8px 8px 16px 4px rgba(133, 139, 146, 0.06);
  border-radius: 20px;
`;

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
`;

const Identifier = styled.div`
  margin-left: 32px;
`;

const PageHeader: React.FC = () => {
  return (
    <Wrapper>
      <Container>
        <Identifier>
          <CompanyIdentifier companyName="Bank Website" />
        </Identifier>
        <ProfileCard />
      </Container>
    </Wrapper>
  );
};

export default PageHeader;
