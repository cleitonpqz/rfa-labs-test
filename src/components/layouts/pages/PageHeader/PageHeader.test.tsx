import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import PageHeader from "./PageHeader";

let container = null;

describe("PageHeader", () => {
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("should match snapshot", () => {
    act(() => {
      render(<PageHeader />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
