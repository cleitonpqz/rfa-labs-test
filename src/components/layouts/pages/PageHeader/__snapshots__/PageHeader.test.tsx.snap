// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should match snapshot 1`] = `"<div class=\\"sc-bdvvaa jWEALH\\">Logo + Id</div>"`;

exports[`PageHeader should match snapshot: "<div>Logo + Id</div>" 1`] = `"<div class=\\"sc-bdvvaa jWEALH\\">Logo + Id</div>"`;
